test: remove LTP

mwasilew opened the pull request:
LTP binaries and scripts can be added in the deriving container. This
container should only include fiotest utilities and a simple test script
example.

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@foundries.io>

doanac commented on test-spec.yml:
```diff
@@ -1,9 +1,26 @@
         command:
-          - /usr/share/fio-tests/ltp.sh
-    # repeat not specified, so runs one time
+          - if [ $(uname -r) = "5.10.9-lmp-standard" ]; then /bin/true; else /bin/false; fi
```
how about we just verify "lmp-standard" is in the output or something? otherwise we'll have to change this for every LMP update and users that copy this will see failed tests?

mwasilew replied:
That's why I asked about parameters. Each build might have a different kernel version. IMHO this should be set from the result of CI build. I can remove it for now.

doanac replied:
the problem is fiotest has no context to set this parameter. its just triggered with a 'run tests' which is triggered by aktualizr which found a new Target. Dynamic things like this need to be able to find a source of truth. For example, this test would need to look at `https://api.foundries.io/projects/lmp/builds/1180/runs/intel-corei7-64/` and somehow determine the kernel from a build artifact.

mwasilew commented:
@doanac I removed the kernel tests and added a simple script that executes some commands as 'tests'. Let's use this as reference. Also README was updated with example on how to extend the container.

doanac commented on tests/smoke.sh:
```diff
@@ -0,0 +1,40 @@
+
+set -o pipefail
+
+TESTS="pwd, lsb_release -a, uname -a, ip a, lscpu, vmstat, lsblk, false"
```
lsb_release isn't in my factory build?

also - the false test?

mwasilew replied:
false to show how the failure is handled. I can remove it. Same with lsb_release (which can be added to container as well)

doanac commented on tests/smoke.sh:
```diff
@@ -0,0 +1,40 @@
+set -o pipefail
+
+TESTS="pwd, lsb_release -a, uname -a, ip a, lscpu, vmstat, lsblk, false"
+LOGS="/tmp/TESTS_$(date +%s)_"
```
Unused variable

mwasilew replied:
I wanted to catch the output of each test, but I'll remove it for now. Catching exit status and log at the same time requires sth like this: https://github.com/Linaro/test-definitions/blob/master/automated/lib/sh-test-lib#L322. I don't think we need it right now.

doanac commented on tests/smoke.sh:
```diff
@@ -0,0 +1,40 @@
+    # shellcheck disable=SC2039
+    local test="$1"
+    test_case_id="$(echo "${test}" | awk '{print $1}')"
+	result_dir="${TEST_DIR}/$(date +%s.%N)-${test_case_id}"
+	mkdir ${result_dir}
```
tabs vs spaces

doanac commented on tests/smoke.sh:
```diff
@@ -0,0 +1,40 @@
+}
+
+failed=0
+while [ -n "${TESTS}" ]; do
+    test_cmd="$(echo "${TESTS}" | awk -F',' '{print $1}')"
```
there's an easier way to iterate over the tests
~~~
IFS=","
for test_cmd in $TESTS ; do
~~~

mwasilew replied:
it was copy paste from test-definitions. I'll fix it.

doanac commented on tests/smoke.sh:
```diff
@@ -0,0 +1,40 @@
+    echo
+    echo "Running ${test_case_id} test..."
+    local exit_code=0
+    eval "${test}" || exit_code="$?"
```
this is totally subjective, but our ci-scripts uses something like this to help group the output of subcommands:
~~~
indent() { 
    sed 's/^/|  /'
    echo "|--" 
}

# then you could do:
eval "${test}" | indent || exit_code="$?"
~~~

Which produces output like:
~~~
Running lsblk test...
|  NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
|  mmcblk0     179:0    0 29.8G  0 disk 
|  |-mmcblk0p1 179:1    0 41.5M  0 part /var/rootdirs/mnt/boot
|  `-mmcblk0p2 179:2    0 29.8G  0 part /sysroot
|--
Exit code: 0
Return code: 0
~~~

doanac reviewed (commented):
just some small stuff

mwasilew commented:
@doanac is there anything more required to get this merged?

doanac commented:
no. i'd approved the changes. i'll just merge it now.
